src/libraries/System.Text.RegularExpressions/tests/System.Text.RegularExpressions.Generators.Tests/RegexGeneratorParserTests.cs

```diff
@@ -38,15 +38,55 @@ partial class C
             Assert.Equal("SYSLIB1041", Assert.Single(diagnostics).Id);
         }
 
+        public static IEnumerable<object[]> Diagnostic_MalformedCtor_MemberData()
+        {
+            const string Pre = "[RegexGenerator";
+            const string Post = "]";
+            const string Middle = "\"abc\", RegexOptions.None, -1, \"extra\"";
+
+            foreach (bool withParens in new[] { false, true })
+            {
+                string preParen = withParens ? "(" : "";
+                string postParen = withParens ? ")" : "";
+                for (int i = 0; i < Middle.Length; i++)
+                {
+                    yield return new object[] { Pre + preParen + Middle.Substring(0, i) + postParen };
```

Comment:
The generator isnt parsing the attribute, right? It just asks for RegexGeneratorAttribute and reads off its properties. So most of this is testing the Roslyn parser isnt it?

Reply:
it's testing we don't fail miserably regardless of what data the roslyn object model gives us.
https://github.com/dotnet/runtime/issues/65008